src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Shell/VSShellServices.cs

```
@@ -11,65 +12,54 @@ namespace Microsoft.VisualStudio.Shell;
+            if (vsAppId is not null)
+            {
+                const int VSAPROPID_IsInCommandLineMode = (int)Microsoft.Internal.VisualStudio.AppId.Interop.__VSAPROPID10.VSAPROPID_IsInCommandLineMode;
+                return ErrorHandler.Succeeded(vsAppId.GetProperty(VSAPROPID_IsInCommandLineMode, out object o)) && o is bool isInCommandLineMode && isInCommandLineMode;
+            }
 
-                    return ErrorHandler.Succeeded(hr)
-                        && result is bool isCommandLineMode
-                        && isCommandLineMode;
-                }
+            return false;
+        }
 
-                bool IsPopulateSolutionCacheMode()
-                {
-                    if (commandLine is null)
-                        return false;
+        static async Task<bool> IsPopulateSolutionCacheModeAsync(IVsService<SVsAppCommandLine, IVsAppCommandLine> commandLineService)
+        {
+            IVsAppCommandLine? commandLine = await commandLineService.GetValueOrNullAsync();
+            if (commandLine is null)
+                return false;
 
-                    int hr = commandLine.GetOption("populateSolutionCache", out int populateSolutionCache, out string commandValue);
+            int hr = commandLine.GetOption("populateSolutionCache", out int populateSolutionCache, out string _);
 
-                    return ErrorHandler.Succeeded(hr)
-                        && Convert.ToBoolean(populateSolutionCache);
-                }
-            },
-            joinableTaskContext.Factory);
+            return ErrorHandler.Succeeded(hr)
+                && Convert.ToBoolean(populateSolutionCache);
+        }
     }
 
     public async Task<bool> IsCommandLineModeAsync(CancellationToken cancellationToken)
     {
-        return (await _initialization.GetValueAsync(cancellationToken)).IsCommandLineMode;
+        return s_isCommandLine is not null && await s_isCommandLine;
```

I don't think `s_isCommandLine` should ever be null here. Consider `Assumes.NotNull` instead, to catch bugs in future if we break that, and to avoid having this start failing silently.

**EDIT** per my other comment, perhaps these don't have to be nullable.

---

No longer nullable.